specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-05-01-preview/containerInstance.json

```diff
@@ -0,0 +1,2611 @@
+                },
+                "state": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The state of the container group. Only valid in response."
+                }
+              }
+            },
+            "diagnostics": {
+              "description": "The diagnostic information for a container group.",
+              "$ref": "#/definitions/ContainerGroupDiagnostics"
+            },
+            "subnetIds": {
+              "type": "array",
+              "description": "The subnet resource IDs for a container group.",
+              "items": {
+                "$ref": "#/definitions/ContainerGroupSubnetId"
+              }
+            },
+            "dnsConfig": {
+              "description": "The DNS config information for a container group.",
+              "$ref": "#/definitions/DnsConfiguration"
+            },
+            "sku": {
+              "description": "The SKU for a container group.",
+              "$ref": "#/definitions/ContainerGroupSku"
+            },
+            "encryptionProperties": {
+              "description": "The encryption properties for a container group.",
+              "$ref": "#/definitions/EncryptionProperties"
+            },
+            "initContainers": {
+              "type": "array",
+              "description": "The init containers for a container group.",
+              "items": {
+                "$ref": "#/definitions/InitContainerDefinition"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "extensions": {
+              "type": "array",
+              "description": "extensions used by virtual kubelet",
+              "items": {
+                "$ref": "#/definitions/DeploymentExtensionSpec"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "confidentialComputeProperties": {
+              "description": "The properties for confidential container group",
+              "$ref": "#/definitions/ConfidentialComputeProperties"
+            },
+            "priority": {
+              "type": "string",
+              "description": "The priority of the container group.",
+              "enum": [
+                "Regular",
+                "Spot"
+              ],
+              "x-ms-enum": {
+                "name": "ContainerGroupPriority",
+                "modelAsString": true
+              }
+            },
+            "containerGroupProfile": {
+              "description": "The reference container group profile properties.",
+              "$ref": "#/definitions/ContainerGroupProfileReferenceDefinition"
+            },
+            "standbyPoolProfile": {
+              "description": "The reference standby pool profile properties.",
+              "$ref": "#/definitions/StandbyPoolProfileDefinition"
+            },
+            "isCreatedFromStandbyPool": {
```

Comment:
>isCreatedFromStandbyPool [](http://example.com/codeflow?start=13&length=24)

ARM recommends boolean/switch properties with a more meaningful enum whenever possible. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. For example, In this case, you can use a creationType enum with values like [FromStandbyPool, N/A]

Reply:
The value of this property will always be either True or False, that is why I set it to Boolean, in case that is not preferred, I can change it to an Enum variable with values as True or False

Reply:
@drishtipeshwani09 Please do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean.

You can follow @razvanbadea-msft 's suggestion above.

Reply:
Currently, as this feature has been already released, it is expected to behave in a certain way by the customers. The variables that I have defined as boolean like isCreatedFromStandbyPool or failContainerGroupCreateOnReuseFailure , are expected to have True or False as their corresponding values. So, the code cannot be modified to use enum variables instead.

Reply:
Please apply this feedback to future APIs and create the spec before the API is released so feedback can be applied.

Reply:
Sure, will keep that in mind. Thank you !!